### PR TITLE
[iOS] Add animation when you tap heartmark in timetable list.

### LINKED ISSUE #204: [iOS] Favorite registration micro-interaction for iOS
## Overview
Implement a micro-interaction animation when tapping the favorite button on the timetable list. The animation should show a heart entering the "Favorites" tab (heart icon) in the bottom navigation from the timetable list's favorite button.

## Links
- Design: https://www.figma.com/design/XUk8WMbKCeIdWD5cz9P9JC/DroidKaigi-2024-App-UI

### PR DESCRIPTION
**Pull request by fumiyasac**
## Issue
- close #204

## Overview

- Enable to get tap position.
  - modify `TimetableCard.swift`
    - In order to calculate the value from GeometryReader, it is supported by assigning DragGesture to the Image element instead of Button.
    - The offset value in the Y-axis direction is subtracted for adjustment (decided by device orientation).
    - modify closure `onTapFavorite: (TimetableItem, CGPoint?) -> Void`.
- Fix TimetableListView structure.  (※ This View element is exceptionally not included in TCA.)
  - modify `TimetableListView.swift`
    - Calculate the offset value in the X-axis and Y-axis direction using GeometryReader.
    - Stack the heartmark elements that will be animated using ZStack.
    - When "isFavorited" is false and the state before ".favoriteTapped(timetableItemWithFavorite)", this view executes animation.

__Points that seem difficult to achieve:__

1. Implementation to cover the Animation element over the TabBar's heartmark.
2. To produce a movement that looks like an arc during animation.

## Note

This is an experimental creation, so feel free to discard it if it doesn't meet your needs 🙇

## Movie

__Portrait:__

https://github.com/user-attachments/assets/e069b552-25ff-458b-abf5-a316eab4e632

__Landscape:__

https://github.com/user-attachments/assets/2f68e217-017a-414f-bf4c-33ca47b66a83

### CODE REVIEW

File: app-ios/Sources/TimetableFeature/TimetableListView.swift [hunk 1]
```diff
@@ -88,22 +88,91 @@ struct TimetableListView: View {
...
+        targetTimetableItemId = timetableItem.id
+
+        // MEMO: Execute animation.
+        if targetTimetableItemId != nil {
+            withAnimation(.easeOut(duration: 1)) {
+                animationProgress = 1
+            }
+            Task {
+                try await Task.sleep(nanoseconds: 1_000_000_000)
+                targetTimetableItemId = nil
```
Comment (ry-itto) [inline app-ios/Sources/TimetableFeature/TimetableListView.swift:172]: Can you fix this warning? 🙏🏼 

> capture of 'self' with non-sendable type 'TimetableListView' in a `@Sendable` closure

Comment (fumiyasac) [reply app-ios/Sources/TimetableFeature/TimetableListView.swift:172]: @ry-itto I think you can solve it like this.

```swift
@MainActor
struct TimetableListView: View { ... }
```

**Comment by fumiyasac** [comment]
@ry-itto I fixed warning. Please check it.

**Review by ry-itto — APPROVED** [review]
LGTM!

**Pull request merged**